src/components/field-types/CodeEditorFieldComponent.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <validate :state="state" :custom="{'validate': isValid}">
+    <div class="form-group">
+      <label :for="field.id">{{ field.label }}</label>
+      <div id="styling" :style="style">
```

Comment:
use the new and awesome bootstrap4 border classes (https://getbootstrap.com/docs/4.0/utilities/borders/)

```
:class="{'border border-danger': isInvalid}"

computed: {
  isInvalid () {
     return !this.isValid || (this.isRequired && this.localValue === '')
  }
}
```